src/string.rs

```diff
@@ -151,23 +155,15 @@ impl ExternalOneByteStringResource {
   }
 
   /// Returns the data owned by this resource as a string slice.
-  /// The data is guaranteed to be ASCII.
-  pub fn as_str(&self) -> &str {
+  /// The data is guaranteed to be Latin-1.
+  pub fn as_bytes(&self) -> &[u8] {
     let len = self.length();
-    if len == 0 {
-      ""
-    } else {
-      // SAFETY: We know this is ASCII and length > 0
-      unsafe {
-        std::str::from_utf8_unchecked(std::slice::from_raw_parts(
-          self.data().cast(),
-          len,
-        ))
-      }
-    }
+    // SAFETY: We know this is Latin-1
+    unsafe { std::slice::from_raw_parts(self.data().cast(), len) }
```

Comment:
Is `self.data()` always non-null?

Reply:
That's actually a good question, and I'm not sure. At least if the length is non-zero then it should always be.

I returned the length check that I had removed to be absolutely sure.